odbc/src/api/functions.rs

```diff
@@ -1785,11 +1785,441 @@ pub unsafe extern "C" fn SQLGetInfo(
+            // TODO: (code review) The docs say:
+            //  - "A character string that contains all special characters (that is,
+            //     all characters except a through z, A through Z, 0 through 9, and
+            //     underscore) that can be used in an identifier name, such as a table
+            //     name, column name, or index name, on the data source. For example,
+            //     "#$^". If an identifier contains one or more of these characters, the
+            //     identifier must be a delimited identifier."
+            //  - MongoSQL's grammar defines an identifier character as [^\x00], as in, literally any non-null character.
+            //    Surely we aren't meant to return a "special characters" string that contains every character, are we?
+        }
+        // SQL_MAX_COLUMNS_IN_GROUP_BY
+        InfoType::MaxColumnsInGroupBy => {
+            // MongoSQL does not have an explicit maximum number of
+            // columns allowed in a GROUP BY clause.
+            return i16_len::set_output_fixed_data(&SQL_U16_ZERO, info_value_ptr, string_length_ptr);
+        }
+        // SQL_MAX_COLUMNS_IN_ORDER_BY
+        InfoType::MaxColumnsInOrderBy => {
+            // MongoSQL does not have an explicit maximum number of
+            // columns allowed in a ORDER BY clause.
+            return i16_len::set_output_fixed_data(&SQL_U16_ZERO, info_value_ptr, string_length_ptr);
+        }
+        // SQL_MAX_COLUMNS_IN_SELECT
+        InfoType::MaxColumnsInSelect => {
+            // MongoSQL does not have an explicit maximum number of
+            // columns allowed in a SELECT clause.
+            return i16_len::set_output_fixed_data(&SQL_U16_ZERO, info_value_ptr, string_length_ptr);
+        }
+        // SQL_TIMEDATE_ADD_INTERVALS
+        InfoType::TimedateAddIntervals
+        // SQL_TIMEDATE_DIFF_INTERVALS
+        | InfoType::TimedateDiffIntervals => {
+            // MongoSQL DATEADD and DATEDIFF support the following intervals.
+            const TIMEDATE_INTERVALS: u32 = SQL_FN_TSI_SECOND
+                | SQL_FN_TSI_MINUTE
+                | SQL_FN_TSI_HOUR
+                | SQL_FN_TSI_DAY
+                | SQL_FN_TSI_WEEK
+                | SQL_FN_TSI_MONTH
+                | SQL_FN_TSI_QUARTER
+                | SQL_FN_TSI_YEAR;
+            return i16_len::set_output_fixed_data(&TIMEDATE_INTERVALS, info_value_ptr, string_length_ptr);
+        }
+        // SQL_CATALOG_LOCATION
+        InfoType::CatalogLocation => {
+            // MongoSQL puts the catalog (database) at the start of a qualified
+            // table name. As in, db.table.
+            return i16_len::set_output_fixed_data(&SQL_CL_START, info_value_ptr, string_length_ptr);
+        }
+        // SQL_SQL_CONFORMANCE
+        InfoType::SqlConformance => {
+            // MongoSQL is SQL-92 Entry level compliant.
+            return i16_len::set_output_fixed_data(
+                &SQL_SC_SQL92_ENTRY,
```

Comment:
Yeah, I don't know if we an Entry level or Intermediate.
We are strictly Entry level because we support case expression (6.9, 2.a) and cast specification (6.10 2.a).
But we might not be strictly Intermediate level either, I did not have time to go over all the restriction on the SQL spec.

Reply:
Left it as Entry for now still.